crates/bitwarden/src/crypto/enc_string.rs

```diff
@@ -62,52 +51,66 @@ impl FromStr for EncString {
+                if parts.len() == 3 {
+                    ("1", parts) // AesCbc128_HmacSha256_B64
+                } else {
+                    ("0", parts) // AesCbc256_B64
+                }
+            }
+        };
+
+        fn from_b64_vec(s: &str) -> Result<Vec<u8>> {
+            Ok(BASE64_ENGINE
+                .decode(s)
+                .map_err(EncStringParseError::InvalidBase64)?)
+        }
+
+        fn from_b64<const N: usize>(s: &str) -> Result<[u8; N]> {
+            Ok(from_b64_vec(s)?.try_into().map_err(invalid_len_error(N))?)
+        }
 
-        let parts: Vec<_> = data.split('|').collect();
         match (enc_type, parts.len()) {
-            ("0", 2) => unimplemented!(),
+            ("0", 2) => {
+                let iv = from_b64(parts[0])?;
+                let data = from_b64_vec(parts[1])?;
 
+                Ok(EncString::AesCbc256_B64 { iv, data })
+            }
             ("1" | "2", 3) => {
-                let iv_str = parts[0];
-                let data_str = parts[1];
-                let mac_str = parts[2];
-
-                let iv = BASE64_ENGINE
-                    .decode(iv_str)
-                    .map_err(EncStringParseError::InvalidBase64)?
-                    .try_into()
-                    .map_err(invalid_len_error(16))?;
-
-                let mac = BASE64_ENGINE
-                    .decode(mac_str)
-                    .map_err(EncStringParseError::InvalidBase64)?
-                    .try_into()
-                    .map_err(invalid_len_error(32))?;
-
-                let data = BASE64_ENGINE
-                    .decode(data_str)
-                    .map_err(EncStringParseError::InvalidBase64)?;
+                let iv = from_b64(parts[0])?;
+                let data = from_b64_vec(parts[1])?;
+                let mac = from_b64(parts[2])?;
 
                 if enc_type == "1" {
                     Ok(EncString::AesCbc128_HmacSha256_B64 { iv, mac, data })
                 } else {
                     Ok(EncString::AesCbc256_HmacSha256_B64 { iv, mac, data })
                 }
             }
-
             ("3" | "4", 1) => {
-                let data = BASE64_ENGINE
-                    .decode(data)
-                    .map_err(EncStringParseError::InvalidBase64)?;
+                let data = from_b64_vec(parts[0])?;
                 if enc_type == "3" {
                     Ok(EncString::Rsa2048_OaepSha256_B64 { data })
                 } else {
                     Ok(EncString::Rsa2048_OaepSha1_B64 { data })
                 }
             }
-            ("5" | "6", 2) => {
-                unimplemented!()
+            #[allow(deprecated)]
+            ("5" | "6", 1) => {
+                let data = from_b64_vec(parts[0])?;
+                if enc_type == "5" {
+                    Ok(EncString::Rsa2048_OaepSha256_HmacSha256_B64 { data })
+                } else {
+                    Ok(EncString::Rsa2048_OaepSha1_HmacSha256_B64 { data })
+                }
```

Comment:
Is this cleaner than just writing

```rust
("5", 1) => {
  Ok(EncString::Rsa2048_OaepSha256_HmacSha256_B64 { data: from_b64_vec(parts[0])? })
}
("6", 1) => {
  Ok(EncString::Rsa2048_OaepSha1_HmacSha256_B64 { data: from_b64_vec(parts[0])? })
}
```

Reply:
I kept it like it was previoulsy, but it's definitely not very useful now that most of the match branches are very short with the `from_b64_vec`. We could even split `("1" | "2", 3)` as well, as it's only three lines now.